atom/browser/api/atom_api_system_preferences_mac.mm

```diff
@@ -78,6 +79,33 @@ static bool FromV8(v8::Isolate* isolate,
 // The map to convert |id| to |int|.
 std::map<int, id> g_id_map;
 
+bool ParseMediaType(const std::string& media_type, AVMediaType* out) {
+  if (media_type == "camera") {
+    *out = AVMediaTypeVideo;
+    return true;
+  } else if (media_type == "microphone") {
+    *out = AVMediaTypeAudio;
+    return true;
+  } else {
+    return false;
+  }
+}
+
+std::string ConvertAuthorizationStatus(AVAuthorizationStatusMac status) {
+  switch (status) {
+    case AVAuthorizationStatusNotDeterminedMac:
+      return "not-determined";
+    case AVAuthorizationStatusRestrictedMac:
+      return "restricted";
+    case AVAuthorizationStatusDeniedMac:
+      return "denied";
+    case AVAuthorizationStatusAuthorizedMac:
+      return "granted";
+    default:
+      return "";
```

Comment:
`NOT_REACHED` maybe?

Reply:
Perhaps `"unknown"`, I wouldn't want existing Electron application to crash / dcheck if Apple added another thing to this enum

Reply:
we don't used it anywhere else in Electron code

Reply:
changed to `"unknown"`